debian/vyos-1x.postinst

```diff
@@ -21,6 +21,12 @@ if ! grep -q '^openvpn' /etc/passwd; then
     adduser --quiet --firstuid 100 --system --group --shell /usr/sbin/nologin openvpn
 fi
 
+# Add 2FA support for SSH
+sudo sed -i '/^@include common-auth/a # Check OTP 2FA, if configured for the user\nauth required pam_google_authenticator.so nullok' /etc/pam.d/sshd
```

Comment:
Do we really need to enable the PAM module all the time? Why not enable it on demand like RADIUS as we do for RADIUS authentication.

Reply:
The idea is that 2FA can be enabled separately for each user
If the user does not have an OTP key configured, then 2FA will not be checked for him
I'm not sure if it is possible to enable the PAM module separately for each user
I agree with RADIUS, it is either enabled or not.
But in this case, separate support for each user is desirable

Reply:
> I'm not sure if it is possible to enable the PAM module separately for each user

There are certain pam condition modules available - all causing more or less headaches.
Please use the debian way as much as possible and enable 2FA if any one user has 2FA configured using pam-auth-update.

Reply:
libpam-google-authenticator does not provide templates:
vyos@vyos# ls /usr/share/pam-configs/
capability mkhomedir radius systemd unix
In addition, pam-auth-update, changes the common-* files (If I understand correctly). I only want to change the files /etc/pam.d/sshd and /etc/pam.d/login. Without affecting other possible methods (like ocserv or ppp)